src/assets/woodpeckerCI/woodpeckerCI.md

```diff
@@ -0,0 +1,254 @@
+      - "traefik.http.routers.woodpecker-secure.tls=true"
+      - "traefik.http.services.woodpecker-secure.loadbalancer.server.port=8000"
+      # HTTP router + redirect to HTTPS
+      - "traefik.http.routers.woodpecker-http.rule=Host(`your-ci-server`)"
+      - "traefik.http.routers.woodpecker-http.entrypoints=web"
+      - "traefik.http.routers.woodpecker-http.middlewares=redirect-to-https"
+      # Redirect middleware
+      - "traefik.http.middlewares.redirect-to-https.redirectscheme.scheme=https"
+    networks:
+      - woodpecker-net
+
+  woodpecker-agent:
+    image: woodpeckerci/woodpecker-agent:v3
+    command: agent
+    restart: always
+    depends_on:
+      - woodpecker-server
+    volumes:
+      - woodpecker-agent-config:/etc/woodpecker
+      - /var/run/docker.sock:/var/run/docker.sock
+    environment:
+      - WOODPECKER_LOG_LEVEL=debug
+      - WOODPECKER_SERVER=woodpecker-server:9000
+      - WOODPECKER_AGENT_SECRET=${WOODPECKER_AGENT_SECRET}
+    networks:
+      - woodpecker-net
+```
+
+## Authenticating with GitHub
+
+To connect GitHub with Woodpecker:
+
+1. Register a new OAuth application in your GitHub developer settings.
+
+2. Set the callback URL to `https://<your-ci-server>/authorize`
+
+3. Copy the generated Client ID and Secret.
+
+4. Use these values in the `WOODPECKER_GITHUB_CLIENT` and `WOODPECKER_GITHUB_SECRET` env vars.
+
+After that, you can log into the Woodpecker web UI using your GitHub account.
+
+![GitHub Login](https://raw.githubusercontent.com/JankariTech/blog/woodpecker-ci-starlark/src/assets/woodpeckerCI/images/github.png)
+
+## Enabling a Repository
+From the Woodpecker UI:
+
+- Click on Add Repository.
+
+![Add Repo](https://raw.githubusercontent.com/JankariTech/blog/woodpecker-ci-starlark/src/assets/woodpeckerCI/images/add.png)
+
+- Enable it.
+
+![Enable Repo](https://raw.githubusercontent.com/JankariTech/blog/woodpecker-ci-starlark/src/assets/woodpeckerCI/images/enable.png)
+
+Woodpecker will automatically add the necessary webhooks to the repo.
+
+
+After enabling your desired repository, there's one more thing that you need to do i.e. to set pipeline config path.
+
+- From the settings of your Woodpecker Server, go to repositories and click on the enabled repo's settings icon as shown below:
+
+![Settings](https://raw.githubusercontent.com/JankariTech/blog/woodpecker-ci-starlark/src/assets/woodpeckerCI/images/settings.png)
+
+- Then, add `.woodpecker.star` as pipeline config path and save it:
+
+![Pipeline Config](https://raw.githubusercontent.com/JankariTech/blog/woodpecker-ci-starlark/src/assets/woodpeckerCI/images/path.png)
+
+## Generating the Public Key
+Every request sent by Woodpecker is signed using a http-signature by a private key (ed25519) generated on the first start of the Woodpecker server. You can get the public key for the verification of the http-signature from:
+```console
+http(s)://your-ci-server/api/signature/public-key
```

Comment:
can it be http with the setup above?
which of the parts in that URL are place-holders and what are real values? They all look the same